hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -136,8 +136,20 @@ def get_type_hints(thing):
         NameError for unresolvable forward references, just return an empty dict.
```

Comment:
Please also update the docstring of this function to describe what it does now, and why we need the additional logic for classes.